domain/tutoringModel/NestorFolder/nestor_training.py

```diff
@@ -0,0 +1,80 @@
+from pgmpy.estimators import ExpectationMaximization as EM
+from pgmpy.readwrite import XMLBIFWriter
+
+from domain.tutoringModel.NestorFolder.nestor_config import (
+    cpd_aam,
+    cpd_bo,
+    cpd_ct,
+    cpd_ex,
+    cpd_lg,
+    cpd_ms,
+    cpd_qu,
+    cpd_su,
+    cpd_tam,
+    cpd_vam,
+    df_data_categories,
+    edges_list_active_reflective,
+    edges_list_only_psy_models,
+    edges_list_sensory_intuitive,
+    edges_list_sequential_global,
+    edges_list_visual_verbal,
+    path_to_trainedmodel,
+    state_names_bn_training_psy_model,
+)
+from domain.tutoringModel.NestorFolder.nestor_utils import (  # nopep8
+    build_bn,
+    extend_bn_to_le,
+    train_bn,
+)
+
+
+def build_train_save_nestor():
+    """This function builds, trains, using utils and
+    saves Causal Nestor model to local directory"""
+    # Building the Topology of BN for Psychological models
+    bn = build_bn(edges_list=edges_list_only_psy_models)
```

Comment:
Argument of type "list[Unknown]" cannot be assigned to parameter "edges_list" of type "int" in function "build_bn"
  "list[Unknown]" is incompatible with "int"

Reply:
In domain.tutoringModel.NesorFolder.nestor_utils.py I changed the default datatype for "build_bn" helper function. 
It must be accepting a list to build the topology of BN. I do not remember, why I defined the default datatype to be integer :-

Reply:
If this is fine, we can close this conversation!